### PR TITLE
Update terraform secrets version to v.0.6.0

### PR DESCRIPTION
**Pull request by robmonte**
Update terraform secrets plugin to v0.6.0.

```
go get github.com/hashicorp/vault-plugin-secrets-terraform/@v0.6.0
go mod tidy
```

### CODE REVIEW

**Review by maxcoulombe — APPROVED** [review]

File: changelog/17172.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
+secrets/terraform: upgrade to v0.6.0
```
Comment (austingebauer) [inline changelog/17172.txt:2]: I don't think we've been including CL entires for the plugin version upgrades like this. The entries are mostly focused on notable bug fixes, improvements, features that a customer would want to know about coming in v0.6.0.

Comment (austingebauer) [reply changelog/17172.txt:2]: If there are none, then feel free to omit the CL entry.

**Review by austingebauer — APPROVED** [review]
LGTM otherwise

File: changelog/17172.txt (hunk 1 above)
Comment (robmonte) [reply changelog/17172.txt:2]: This plugin has no changes, would it be better to remove the changelog file entirely? I can fix this on my kerberos update PR which does have changes.

Comment (robmonte) [reply changelog/17172.txt:2]: Answered before I asked 😆

Comment (austingebauer) [reply changelog/17172.txt:2]: Yep, feel free to remove it 👍  Technically, nothing has changed in the plugin that a user should know about.

**Pull request merged**